build/azure-pipelines/web/sql-product-build-web.yml

```diff
@@ -44,59 +32,175 @@ steps:
     node build/azure-pipelines/mixin
   displayName: Mix in quality
 
 - script: |
     set -e
-    VSCODE_MIXIN_PASSWORD="$(github-distro-mixin-password)" \
-      yarn gulp vscode-web-min-ci
-  displayName: Build
+    yarn sqllint
+    yarn gulp hygiene
+    yarn strict-vscode
+    yarn valid-layers-check
+  displayName: Run hygiene, eslint
+  condition: and(succeeded(), eq(variables['VSCODE_STEP_ON_IT'], 'false'))
 
-  # upload only the workbench.web.api.js source maps because
-  # we just compiled these bits in the previous step and the
-  # general task to upload source maps has already been run
 - script: |
     set -e
-    AZURE_STORAGE_ACCOUNT="$(sourcemap-storage-account)" \
-    AZURE_STORAGE_ACCESS_KEY="$(sourcemap-storage-key)" \
-      node build/azure-pipelines/upload-sourcemaps out-vscode-web-min out-vscode-web-min/vs/workbench/workbench.web.api.js.map
-  displayName: Upload sourcemaps (Web)
+    yarn gulp compile-build
+    yarn gulp compile-extensions-build
+    yarn gulp minify-vscode
+    yarn gulp vscode-linux-x64-min-ci
+    yarn gulp vscode-web-min-ci
+    yarn gulp vscode-reh-linux-x64-min
+    yarn gulp vscode-reh-web-linux-x64-yarnrc-extensions
+    yarn gulp vscode-reh-web-linux-x64-min
+  displayName: Compile
 
 # - script: |
 #     set -e
-#     AZURE_DOCUMENTDB_MASTERKEY="$(builds-docdb-key-readwrite)" \
-#     AZURE_STORAGE_ACCESS_KEY_2="$(vscode-storage-key)" \
-#     VSCODE_MIXIN_PASSWORD="$(github-distro-mixin-password)" \
-#     ./build/azure-pipelines/web/publish.sh
-#   displayName: Publish
+#     AZURE_STORAGE_ACCOUNT="$(sourcemap-storage-account)" \
+#     AZURE_STORAGE_ACCESS_KEY="$(sourcemap-storage-key)" \
+#       node build/azure-pipelines/upload-sourcemaps
+#   displayName: Upload sourcemaps
 
+- script: |
+    set -e
+
+    VERSION=$(node -p "require(\"./package.json\").version")
+
+    echo -e "{  \"version\": \"$VERSION\", \"quality\": \"$VSCODE_QUALITY\", \"commit\": \"$BUILD_SOURCEVERSION\" }" > ".build/version.json"
+
+    node build/azure-pipelines/common/copyArtifacts.js
+  displayName: Write Version Information
+
+- task: PublishBuildArtifacts@1
+  displayName: 'Publish Artifact: drop'
+
+- script: |
+    set -e
+    tar -czf $(Build.ArtifactStagingDirectory)/compilation.tar.gz .build out-*
+  displayName: Compress compilation artifact
+
+- task: PublishPipelineArtifact@1
+  inputs:
+    targetPath: $(Build.ArtifactStagingDirectory)/compilation.tar.gz
+    artifactName: Compilation
+  displayName: Publish compilation artifact
+
+- script: |
+    set -e
+    yarn gulp package-rebuild-extensions
+    yarn gulp compile-extensions
+    yarn gulp package-external-extensions
+  displayName: Package External extensions
+
+- script: |
+    set -e
+    yarn gulp package-langpacks
+  displayName: Package Langpacks
+
+- script: |
+    set -e
+    yarn gulp vscode-linux-x64-build-deb
+  displayName: Build Deb
+
+- script: |
+    set -e
+    yarn gulp vscode-linux-x64-build-rpm
+  displayName: Build Rpm
+
```

Comment:
Is there any reason why we want to build deb/rpm in web build as well? I thought we had that in normal linux build already.

Reply:
I setup the build to produce all the linux artifacts.  primary reason is if I want to compare functionality in web vs. desktop I have all the artifacts together.  perhaps we can remove in the future if we want to save build time, storage space, etc.  but I'd like to keep that enabled for time being.

Reply:
Got it, this makes sense!

Reply:
Maybe add a comment about that here so others know why in case someone wants to remove it later on